src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -848,6 +892,9 @@ ExecuteSingleModifyTask(CitusScanState *scanState, Task *task, bool multipleTask
 		 */
 		if (!gotResults && expectResults)
 		{
+			/* DDL tasks don't need to store any results */
+			Assert(task->taskType != DDL_TASK);
```

Comment:
is this universally true? We wouldn't propagate them, but e.g. EXECUTE commands may return results.